react-form-controls/my-app/src/RegistrationFormControlled.js

```diff
@@ -0,0 +1,32 @@
+import { useState } from "react";
+
+export default function RegistrationFormControlled() {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    console.log(username, password);
+  }
+
+  return (
+    <form method="post" onSubmit={handleSubmit}>
+
+      <label>
+        UserName: <input onChange={(e) => setUsername(e.target.value)} type="text" name="userName" />
```

Comment:
The `name` isn't required if it's uncontrolled. The `value` is required, though. Obviously, you've figured out that occasionally it works, but it is buggy.